tracing-serde/src/lib.rs

```diff
@@ -15,6 +15,33 @@ use tracing_core::{
     span::{Attributes, Id, Record},
 };
 
+/// A bridge between `fmt::Write` and `io::Write`.
+pub struct WriteAdaptor<'a> {
+    fmt_write: &'a mut dyn fmt::Write,
+}
```

Comment:
Nice! Could you explain in the doc comment as to to why a bridge is necessary between `fmt::Write` and `io::Write`? Something along the lines of: 

> This is needed because `tracing-subscriber`'s FormatEvent expects a `fmt::Write` while `serde_json`'s `Serializer` expects an `io::Write`.

Reply:
Or then make it private in subscriber. I think it's kind of useful struct though...